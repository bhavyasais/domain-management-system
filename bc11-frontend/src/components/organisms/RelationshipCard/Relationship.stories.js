import React, { useState } from 'react';
import BarGraph from '../../molecules/BarGraph/BarGraph';
import RelationshipCard from './RelationshipCard';

export default {
  title: 'Organisms/RelationshipCard',
};

export const relationshipCard = () => {
  const [value1, setValue1] = useState({ name: '', value: '', data: null });
  const [value2, setValue2] = useState({ name: '', value: '', data: null });
  const list = [
    {
      value: 'zemoso',
      name: 'zemoso',
      data: [
        { date: '1st Dec', value: 30 },
        { date: '2nd Dec', value: 5 },
        { date: '3rd Dec', value: 20 },
        { date: '4th Dec', value: 40 },
        { date: '5th Dec', value: 12 },
        { date: '6th Dec', value: 25 },
        { date: '7th Dec', value: 30 },
      ],
    },
    {
      value: 'stanford',
      name: 'stanford',
      data: [
        { date: '1st Dec', value: 30 },
        { date: '2nd Dec', value: 15 },
        { date: '3rd Dec', value: 12 },
        { date: '4th Dec', value: 40 },
        { date: '5th Dec', value: 20 },
        { date: '6th Dec', value: 5 },
        { date: '7th Dec', value: 30 },
      ],
    },
  ];

  const list1 = [
    {
      value: 'zemoso',
      name: 'zemoso',
      data: [
        { date: '1st Dec', value: 30 },
        { date: '2nd Dec', value: 5 },
        { date: '3rd Dec', value: 20 },
        { date: '4th Dec', value: 40 },
        { date: '5th Dec', value: 12 },
        { date: '6th Dec', value: 25 },
        { date: '7th Dec', value: 30 },
        { date: '8th Dec', value: 40 },
        { date: '9th Dec', value: 12 },
        { date: '10th Dec', value: 25 },
        { date: '11th Dec', value: 30 },
        { date: '12th Dec', value: 25 },
        { date: '13th Dec', value: 30 },
        { date: '14th Dec', value: 25 },
        { date: '15th Dec', value: 30 },
        { date: '16th Dec', value: 25 },
        { date: '17th Dec', value: 30 },
      ],
    },
    {
      value: 'stanford',
      name: 'stanford',
      data: [
        { date: '1st Dec', value: 30 },
        { date: '2nd Dec', value: 5 },
        { date: '3rd Dec', value: 20 },
        { date: '4th Dec', value: 40 },
        { date: '5th Dec', value: 12 },
        { date: '6th Dec', value: 25 },
        { date: '7th Dec', value: 30 },
        { date: '8th Dec', value: 40 },
        { date: '9th Dec', value: 12 },
        { date: '10th Dec', value: 25 },
        { date: '11th Dec', value: 30 },
        { date: '12th Dec', value: 25 },
        { date: '13th Dec', value: 30 },
        { date: '14th Dec', value: 25 },
        { date: '15th Dec', value: 30 },
        { date: '16th Dec', value: 25 },
        { date: '17th Dec', value: 30 },
      ],
    },
  ];

  const handleGroupChange = (value) => {
    console.log(value);
    setValue1(list.find((item) => item.value === value));
  };
  const handleTrustscoreChnage = (value) => {
    console.log(value);
    setValue2(list1.find((item) => item.value === value));
  };

  return (
    <RelationshipCard header="Relationship">
      <BarGraph
        header="Relationship vs Group"
        list={list1}
        selectedValue={value1}
        onDropdownChange={handleGroupChange}
      />
      <BarGraph
        header="Relationship vs Trustscore Range"
        list={list1}
        selectedValue={value2}
        onDropdownChange={handleTrustscoreChnage}
      />
    </RelationshipCard>
  );
};
