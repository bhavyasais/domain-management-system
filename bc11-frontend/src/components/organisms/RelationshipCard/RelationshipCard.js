import { Grid, Typography, withStyles } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import BarGraph from '../../molecules/BarGraph/BarGraph';
import Dropdown from '../../molecules/Dropdown/Dropdown';
import clsx from 'clsx';
import { RELATIONSHIP_CARD_LEGEN_2, RELATIONSHIP_CARD_LEGEN_1 } from '../../../utils/Strings';

const styles = () => ({
  containerStyle: {
    border: 'solid 1px #eaedf3',
    margin: 0,
    width: '100%',
  },
  headerStyle: {
    borderBottom: '1px solid #eeeeee',
    width: '100%',
  },
  headerText: {
    margin: '12px 24px 12px 24px',
  },
  circle: {
    height: '8px',
    width: '8px',
    borderRadius: '50%',
    marginRight: '4px',
  },
  legendContainer: {
    paddingLeft: '24px',
  },
  dropDown: {
    marginRight: '36px',
  },
});

const RelationshipCard = ({ classes, header, children, containerStyle }) => {
  return (
    <Grid item container spacing={2} className={clsx(classes.containerStyle, containerStyle)}>
      {header && (
        <Grid item container className={classes.headerStyle} alignItems="center">
          <Typography variant="h6" className={classes.headerText}>
            {header}
          </Typography>
        </Grid>
      )}
      <Grid item container className={classes.legendContainer}>
        <Grid item container direction="row">
          <Grid
            item
            container
            direction="row"
            alignItems="center"
            xs={7}
            style={{ padding: '24px 24px 12px 24px', color: '#949494' }}
          >
            <div className={classes.circle} style={{ backgroundColor: '#ff3162' }} />
            <Typography variant="body1" style={{ marginRight: '20px' }}>
              {RELATIONSHIP_CARD_LEGEN_1}
            </Typography>
            <div className={classes.circle} style={{ backgroundColor: '#ffab2b' }} />
            <Typography variant="body1">{RELATIONSHIP_CARD_LEGEN_2}</Typography>
          </Grid>
          <Grid item container direction="row" alignItems="center" xs={5} justify="flex-end">
            <Typography variant="body1" style={{ color: '#949494' }}>
              Showing for:
            </Typography>
            <Dropdown containerStyle={classes.dropDown} />
          </Grid>
        </Grid>
      </Grid>
      <Grid
        container
        item
        direction="row"
        spacing={4}
        xs={12}
        style={{ padding: '12px 24px 12px 24px', width: '100%' }}
      >
        {children?.map((item) => (
          <Grid item container xs={6}>
            {item}
          </Grid>
        ))}
      </Grid>
    </Grid>
  );
};

RelationshipCard.propTypes = {
  header: PropTypes.string,
  children: PropTypes.arrayOf(PropTypes.element),
};
RelationshipCard.defaultProps = {
  header: null,
  children: null,
};

export default withStyles(styles)(RelationshipCard);
