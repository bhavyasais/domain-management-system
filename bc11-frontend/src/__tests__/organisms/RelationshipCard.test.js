import React from 'react';
import renderer from 'react-test-renderer';
import RelationshipCard from '../../components/organisms/RelationshipCard/RelationshipCard';

describe('<SideNavigationBar/>', () => {
  let mounted;
  let onClick;
  const list = [
    {
      value: 'zemoso',
      name: 'zemoso',
      data: [
        { date: '1st Dec', value: 30 },
        { date: '2nd Dec', value: 5 },
        { date: '3rd Dec', value: 20 },
        { date: '4th Dec', value: 40 },
        { date: '5th Dec', value: 12 },
        { date: '6th Dec', value: 25 },
        { date: '7th Dec', value: 30 },
      ],
    },
    {
      value: 'stanford',
      name: 'stanford',
      data: [
        { date: '1st Dec', value: 30 },
        { date: '2nd Dec', value: 15 },
        { date: '3rd Dec', value: 12 },
        { date: '4th Dec', value: 40 },
        { date: '5th Dec', value: 20 },
        { date: '6th Dec', value: 5 },
        { date: '7th Dec', value: 30 },
      ],
    },
  ];

  it('Renders correctly', () => {
    const wrapper = renderer.create(
      <RelationshipCard
        groupList={list}
        trustscoreList={list}
        groupSelectedValue={{ name: '', value: '', data: null }}
        trustscoreSelectedValue={{ name: '', value: '', data: null }}
        onGroupChange={onClick}
        onTrustscoreChange={onClick}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
