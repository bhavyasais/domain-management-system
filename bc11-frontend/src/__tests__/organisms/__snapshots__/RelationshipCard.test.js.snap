// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavigationBar/> Renders correctly 1`] = `
<div
  className="MuiGrid-root RelationshipCard-containerStyle-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
>
  <div
    className="MuiGrid-root RelationshipCard-legendContainer-5 MuiGrid-container MuiGrid-item"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-7"
        style={
          Object {
            "color": "#949494",
            "padding": "24px 24px 12px 24px",
          }
        }
      >
        <div
          className="RelationshipCard-circle-4"
          style={
            Object {
              "backgroundColor": "#ff3162",
            }
          }
        />
        <p
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "marginRight": "20px",
            }
          }
        >
           Lorem ipsum dolor sit amet, consectetur adipiscing elit
        </p>
        <div
          className="RelationshipCard-circle-4"
          style={
            Object {
              "backgroundColor": "#ffab2b",
            }
          }
        />
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
           Lorem ipsum dolor sit amet, consectetur adipiscing elit
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-5"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "color": "#949494",
            }
          }
        >
          Showing for:
        </p>
        <div
          className="RelationshipCard-dropDown-6"
        >
          <div
            className="MuiFormControl-root Dropdown-formControl-7"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root Dropdown-dropdown-9 MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="simple-select-label simple-select"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="simple-select"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                None
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-11"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-item MuiGrid-grid-xs-12"
    style={
      Object {
        "padding": "12px 24px 12px 24px",
        "width": "100%",
      }
    }
  />
</div>
`;
